lib/sequel/plugins/uuid.rb

```diff
@@ -0,0 +1,89 @@
+# frozen-string-literal: true
+
+##
+# LOOKER CUSTOM NOTE
+# -----------------------------------------------------------------------------
+# This file has been manually ported from  a future sequel version to our
+# custom 4.2.0 version of Sequel and is, therefore, not officially supported
+# in this version.
+#
+# Modifications that have been made to support this version are:
+#   * Changed set_column_value, which is not defined in this version, to send
+#   * Change _before_validation to before_validation
+#
+# - Jim < jim@looker.com >
+##
+
+
+require 'securerandom'
+
+module Sequel
+  module Plugins
+    # The uuid plugin creates hooks that automatically create a uuid for every
+    # instance.  Note that this uses SecureRandom.uuid to create UUIDs, and
+    # that method is not defined on ruby 1.8.7.  If you would like to use this
+    # on ruby 1.8.7, you need to override the Model#create_uuid private method
+    # to return a valid uuid.
+    #
+    # Usage:
+    #
+    #   # Uuid all model instances using +uuid+
+    #   # (called before loading subclasses)
+    #   Sequel::Model.plugin :uuid
+    #
+    #   # Uuid Album instances, with custom column name
+    #   Album.plugin :uuid, :field=>my_uuid
+    module Uuid
+      # Configure the plugin by setting the available options.  Note that
+      # if this method is run more than once, previous settings are ignored,
+      # and it will just use the settings given or the default settings.  Options:
+      # :field :: The field to hold the uuid (default: :uuid)
+      # :force :: Whether to overwrite an existing uuid (default: false)
+      def self.configure(model, opts=OPTS)
+        model.instance_eval do
+          @uuid_field = opts[:field]||:uuid
+          @uuid_overwrite = opts[:force]||false
+        end
+      end
+
+      module ClassMethods
+        # The field to store the uuid
+        attr_reader :uuid_field
+
+        # Whether to overwrite the create uuid if it already exists
+        def uuid_overwrite?
+          @uuid_overwrite
+        end
+
+        Plugins.inherited_instance_variables(self, :@uuid_field=>nil, :@uuid_overwrite=>nil)
+      end
+
+      module InstanceMethods
+        private
+
+        # Set the uuid when creating
+        def before_validation
+          set_uuid if new?
+          super
+        end
+
+        # Create a new UUID.  This method can be overridden to use a separate
+        # method for creating UUIDs.  Note that this method does not work on
+        # ruby 1.8.7, you will have to override it if you are using ruby 1.8.7.
```

Comment:
since this is our own custom version, it might make more sense to just remove the comments related to 1.8.7 since that doesn't matter for us...  Or is this left in here to reduce the amount of possible merge conflicts in the future?